lib/fhir/resources/questionnaire-response.js

```diff
@@ -55,19 +54,7 @@ module.exports = (mongo) => {
           return callback(null, fhirCommon.buildHTTPOutcome(400, 'error', 'invalid', badRequest.message))
         }
 
-        let query = { $and: [] }
-
-        if (queryObject['encounter']) {
-          query['$and'].push({ 'encounter.reference': queryUtils.paramAsReference(queryObject['encounter'][constants.NO_MODIFER], 'Encounter') })
-        }
-
-        if (queryObject['patient']) {
-          query['$and'].push({ 'subject.reference': queryUtils.paramAsReference(queryObject['patient'][constants.NO_MODIFER], 'Patient') })
-        }
-
-        if (queryObject['questionnaire']) {
-          query['$and'].push({ 'questionnaire.reference': queryUtils.paramAsReference(queryObject['questionnaire'][constants.NO_MODIFER], 'Questionnaire') })
-        }
+        const query = queryBuilder.buildQuery(ctx.resourceType, queryObject)
```

Comment:
Lets add in the customParams filter here for completeness, There are no custom params at the moment, but possible in future
`const filteredQueryObject = queryUtils.filterOutProperties(queryObject, Object.keys(customSupportedParams))`